reasoning_gym/dataset.py

```diff
@@ -56,6 +56,8 @@ def score_answer(self, answer: Optional[str], entry: Dict[str, any]) -> float:
         oracle_answer = entry["answer"]
         reward = 0.0
         if answer is not None:
+            if oracle_answer.strip() == oracle_answer:
```

Comment:
I already considered editing the extract-answer regex .. https://github.com/open-thought/reasoning-gym/blob/979b6ba4efe2aa8bb4ef951d059fd8514e551006/reasoning_gym/utils.py#L20 
The problem with completely ignoring white-spaces is that RL could generate many many whitespaces without getting feedback in form of less reward.
We could ignore a single newline/whitespace at beginning and end of answers. Or we add a better `oracle_answer in answer` case .. for example:  `len(oracle_answer) / len(answer)` could make sense.